jmclient/jmclient/wallet.py

```diff
@@ -1032,6 +999,494 @@ def close(self):
     def __del__(self):
         self.close()
 
+class PSBTWalletMixin(object):
+    """
+    Mixin for BaseWallet to provide BIP174
+    functions.
+    """
+    def __init__(self, storage, **kwargs):
+        super(PSBTWalletMixin, self).__init__(storage, **kwargs)
+
+    def is_input_finalized(self, psbt_input):
+        """ This should be a convenience method in python-bitcointx.
+        However note: this is not a static method and tacitly
+        assumes that the input under examination is of the wallet's
+        type.
+        """
+        assert isinstance(psbt_input, btc.PSBT_Input)
+        if not psbt_input.utxo:
+            return False
+        if isinstance(self, (LegacyWallet, SegwitLegacyWallet)):
+            if not psbt_input.final_script_sig:
+                return False
+        if isinstance(self, (SegwitLegacyWallet, SegwitWallet)):
+            if not psbt_input.final_script_witness:
+                return False
+        return True
+
+    @staticmethod
+    def hr_psbt(in_psbt):
+        """ Returns a jsonified indented string with all relevant
+        information, in human readable form, contained in a PSBT.
+        Warning: the output can be very verbose in certain cases.
+        """
+        assert isinstance(in_psbt, btc.PartiallySignedTransaction)
+        outdict = {}
+        outdict["psbt-version"] = in_psbt.version
+
+        # human readable serialization of these three global fields is for
+        # now on a "best-effort" basis, i.e. just takes the representation
+        # provided by the underlying classes in bitcointx, though this may
+        # not be very readable.
+        # TODO: Improve proprietary/unknown as needed.
+        if in_psbt.xpubs:
+            outdict["xpubs"] = {str(k): bintohex(
+                v.serialize()) for k, v in in_psbt.xpubs.items()}
+        if in_psbt.proprietary_fields:
+            outdict["proprietary-fields"] = str(in_psbt.proprietary_fields)
+        if in_psbt.unknown_fields:
+            outdict["unknown-fields"] = str(in_psbt.unknown_fields)
+
+        outdict["unsigned-tx"] = btc.hrt(in_psbt.unsigned_tx, jsonified=False)
+        outdict["psbt-inputs"] = []
+        for inp in in_psbt.inputs:
+            outdict["psbt-inputs"].append(PSBTWalletMixin.hr_psbt_in(inp))
+        outdict["psbt-outputs"] = []
+        for out in in_psbt.outputs:
+            outdict["psbt-outputs"].append(PSBTWalletMixin.hr_psbt_out(out))
+        return json.dumps(outdict, indent=4)
+
+    @staticmethod
+    def hr_psbt_in(psbt_input):
```

Comment:
Same suggestion as above to make these names more descriptive: for example `human_readable_psbt_in`

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/536/commits/19249f7324910c20e9c76ddb3172500520d6d78b